event-adapter/pkg/apiserver/installer/installer.go

```diff
@@ -186,85 +184,36 @@ func (a *MetricsAPIInstaller) registerResourceHandlers(storage rest.Storage, ws
 		}
 
 		ctx = request.WithUserAgent(ctx, req.HeaderParameter("User-Agent"))
-
-		// inject the resource, subresource, and name here so that
-		// we don't have to write custom handler logic
+		name := req.PathParameter("name")
 		resource := req.PathParameter("resource")
-		subresource := req.PathParameter("subresource")
-		ctx = specificcontext.WithResourceInformation(ctx, resource, subresource)
+		ctx = specificcontext.WithResourceInformation(ctx, resource, name)
 
 		return ctx
 	}
 
 	scope := mapping.Scope
-	nameParam := ws.PathParameter("name", "name of the described resource").DataType("string")
+	nameParam := ws.PathParameter("name", "name of the described event").DataType("string")
 	resourceParam := ws.PathParameter("resource", "the name of the resource").DataType("string")
-	subresourceParam := ws.PathParameter("subresource", "the name of the subresource").DataType("string")
-
-	// metrics describing non-namespaced objects (e.g. nodes)
-	rootScopedParams := []*restful.Parameter{
-		resourceParam,
-		nameParam,
-		subresourceParam,
-	}
-	rootScopedPath := "{resource}/{name}/{subresource:*}"
-
-	// metrics describing namespaced objects (e.g. pods)
 	namespaceParam := ws.PathParameter(scope.ArgumentName(), scope.ParamDescription()).DataType("string")
+
 	namespacedParams := []*restful.Parameter{
-		namespaceParam,
-		resourceParam,
 		nameParam,
-		subresourceParam,
-	}
-	// METRIC REGEXP:
-	namespacedPath := scope.ParamName() + "/{" + scope.ArgumentName() + "}/{resource}/{name}/{subresource:*}"
-	namespacedPathPrefix := gpath.Join(a.prefix, scope.ParamName()) + "/"
-	itemPathFn := func(name, namespace, resource, subresource string) bytes.Buffer {
-		var buf bytes.Buffer
-		buf.WriteString(namespacedPathPrefix)
-		buf.WriteString(url.QueryEscape(namespace))
-		buf.WriteString("/")
-		buf.WriteString(url.QueryEscape(resource))
-		buf.WriteString("/")
-		buf.WriteString(url.QueryEscape(name))
-		buf.WriteString("/")
-		buf.WriteString(url.QueryEscape(subresource))
-		return buf
-	}
-
-	namespaceSpecificPath := scope.ParamName() + "/{" + scope.ArgumentName() + "}/metrics/{name:*}"
-	namespaceSpecificParams := []*restful.Parameter{
+		resourceParam,
 		namespaceParam,
-		nameParam,
 	}
-	namespaceSpecificItemPathFn := func(name, namespace, resource, subresource string) bytes.Buffer {
+
+	//EVENT REGEXP:
+	namespacedPath := scope.ParamName() + "/{" + scope.ArgumentName() + "}/{resource}/{name}"
```

Comment:
I'm not sure if this is what you want. Since the only resource you support is Event, something like this might be better:
namespacedPath := scope.ParamName() + "/{" + scope.ArgumentName() + "}/events/{name}"
This explains why you get internal error for paths like "/namespaces/default/evento/foo". I'm not sure how difficult it is to fix this. I suggest you to spend no more than 1h trying to fix this, and if you can't just leave a TODO or comment explaining why you can't use the path I suggested. Also I think you need more test cases to cover other correct resources, for example: "/namespaces/default/pods" and "/namespaces/default/pods/foo".